configure.ac

```diff
@@ -45,9 +49,12 @@ AS_IF([test "$P4_PREFIX" = ""], [
   AC_SUBST([P4_PREFIX])
 ])
 
-diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/test.p4)
-AM_CONDITIONAL([BMV2_TEST_ENV],
-  [test "$P4_NAME" = "test" && test "$P4_PREFIX" = "test" && test "$diff_w_test" = "no"])
+diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/pdtest/pdtest.p4)
+
+if test "$diff_w_test" = "yes" ; then
+    diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/openflow/oftest.p4)
+fi
+AM_CONDITIONAL([BMV2_TEST_ENV], [test "$P4_NAME" = "pdtest" || test "$P4_NAME" = "oftest" && test "$P4_PREFIX" = "pdtest" || test "$P4_PREFIX" = "oftest" && test "$diff_w_test" = "no"])
```

Comment:
I like that because it will make it easier to add more plugins in the future

On Wed, Jun 22, 2016 at 3:57 PM, Antonin Bas <notifications@github.com>
wrote:

> In configure.ac
> <https://github.com/p4lang/p4-build/pull/5#discussion_r68150280>:
>
> > @@ -45,9 +49,12 @@ AS_IF([test "$P4_PREFIX" = ""], [
> >    AC_SUBST([P4_PREFIX])
> >  ])
> >
> > -diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/test.p4)
> > -AM_CONDITIONAL([BMV2_TEST_ENV],
> > -  [test "$P4_NAME" = "test" && test "$P4_PREFIX" = "test" && test "$diff_w_test" = "no"])
> > +diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/pdtest/pdtest.p4)
> > +
> > +if test "$diff_w_test" = "yes" ; then
> > +    diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/openflow/oftest.p4)
> > +fi
> > +AM_CONDITIONAL([BMV2_TEST_ENV], [test "$P4_NAME" = "pdtest" || test "$P4_NAME" = "oftest" && test "$P4_PREFIX" = "pdtest" || test "$P4_PREFIX" = "oftest" && test "$diff_w_test" = "no"])
>
> I thought a little bit about it, and I think it maybe more readable to
> define 2 conditionals instead of one. Something like this:
>
> diff_w_pdtest=
> AM_CONDITIONAL([BMV2_PDTEST_ENV], [test "$P4_NAME" = "test" &&...])
> diff_w_oftest=
> AM_CONDITIONAL([BMV2_OFTEST_ENV], [test "$want_of" = yes && test "$P4_NAME" = "test" &&...])
>
> This would make tests/Makefile.am pretty nice to:
>
> if BMV2_PDTEST_ENV
> SUBDIRS = pdtest
> endif
>
> if BMV2_ODTEST_ENV
> SUBDIRS = oftest
> endif
>
> if !BMV2_PDTEST_ENV
> if !BMV2_ODTEST_ENV
> check-local: ... // the error message
> endif
> endif
>
> What do you think?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/p4lang/p4-build/pull/5/files/bc6ebbd5168c608c6ef0096cf45b26f409ba8efc#r68150280>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AFgBcZ9oo0dm3J0DQ6OSmozKVuYCVvAnks5qOb3VgaJpZM4Iylfb>
> .
>